distribution/lib/Standard/Database/0.0.0-dev/src/Connection/Connection.enso

```diff
@@ -309,6 +352,98 @@ type Connection
     drop_table self table_name if_exists=False =
         self.execute_update (self.dialect.generate_sql (Query.Drop_Table table_name if_exists))
 
+    ## PRIVATE
+       Returns the base `Connection` instance.
+
+       Used, so that all internal helper functions do not need to be replicated
+       on the 'subclasses'.
+    base_connection : Connection
+    base_connection self = self
+
+    ## PRIVATE
+       If no thread (including the current one) is currently running operations
+       on the connection, maintenance will be performed.
+
+       Currently, this consists of removing dry run tables that are no longer
+       used.
+
+       This method should be run by most database operations to ensure that
+       unused tables are cleaned at some point.
+
+       All errors are swallowed and not propagated, so it is safe to call this
+       method wherever. There is no point of calling this method inside of
+       critical sections as then it will not do anything.
+    maybe_run_maintenance self =
+        callback _ =
+            Hidden_Table_Registry.run_maintenance_table_cleanup self
+        self.jdbc_connection.run_maintenance_action_if_possible callback
+
+    ## PRIVATE
+    max_table_name_length : Integer | Nothing
+    max_table_name_length self =
+        reported = self.jdbc_connection.with_metadata .getMaxTableNameLength
+        if reported == 0 then Nothing else reported
+
+    ## PRIVATE
+       Generates a temporary table name for the given table name, used for dry
+       runs.
+
+       The table name is 'stable', meaning that the same name will be returned
+       for the given input `table_name` on subsequent calls, unless the user
+       creates a clashing table in the meantime.
+
+       The table name is guaranteed to be unique for the database at the time it
+       is generated - this is used to ensure that the dry run tables never
+       overwrite pre-existing user data.
+
+       ! Safety
+
+         It is safe to drop/overwrite the table returned by this method, as it
+         can be assumed that it was not created by the user. It either does not
+         (yet) exist, or if it exists, it is present in our hidden table
+         registry - but that means it was created by Enso as a hidden table.
+    generate_dry_run_table_name : Text -> Text
+    generate_dry_run_table_name self table_name =
+        go ix =
+            prefix = "enso-dry-run-" + if ix == 0 then "" else ix.to_text + "-"
+            max_length = (self.max_table_name_length.if_nothing 60) - 1
```

Comment:
If the caller does use a `table_name` that's too long, could this result in a name clash, with one table replacing the other one?

Reply:
The method truncates the name to a safe length and checks if such a table already exists - so no risk of clash.

The only risk is that the method may loop infinitely if the max name is too short and all table names of this format are already taken. I will fix so that it will fail instead.